packages/techdocs-common/src/stages/publish/openStackSwift.ts

```diff
@@ -0,0 +1,260 @@
+    // and the storage client will work.
+    storageClient.getContainer(containerName, (err, container) => {
+      if (container) {
+        logger.info(
+          `Successfully connected to the OpenStack Swift container ${containerName}.`,
+        );
+      } else {
+        logger.error(
+          `Could not retrieve metadata about the OpenStack Swift container ${containerName}. ` +
+            'Make sure the container exists. Also make sure that authentication is setup either by ' +
+            'explicitly defining credentials and region in techdocs.publisher.openStackSwift in app config or ' +
+            'by using environment variables. Refer to https://backstage.io/docs/features/techdocs/using-cloud-storage',
+        );
+
+        logger.error(`from OpenStack client library: ${err.message}`);
+      }
+    });
+
+    return new OpenStackSwiftPublish(storageClient, containerName, logger);
+  }
+
+  constructor(
+    private readonly storageClient: storage.Client,
+    private readonly containerName: string,
+    private readonly logger: Logger,
+  ) {
+    this.storageClient = storageClient;
+    this.containerName = containerName;
+    this.logger = logger;
+  }
+
+  /**
+   * Upload all the files from the generated `directory` to the OpenStack Swift container.
+   * Directory structure used in the bucket is - entityNamespace/entityKind/entityName/index.html
+   */
+  async publish({ entity, directory }: PublishRequest): Promise<void> {
+    try {
+      // Note: OpenStack Swift manages creation of parent directories if they do not exist.
+      // So collecting path of only the files is good enough.
+      const allFilesToUpload = await getFileTreeRecursively(directory);
+      const limiter = createLimiter(10);
+      const uploadPromises: Array<Promise<unknown>> = [];
+      for (const filePath of allFilesToUpload) {
+        // Remove the absolute path prefix of the source directory
+        // Path of all files to upload, relative to the root of the source directory
+        // e.g. ['index.html', 'sub-page/index.html', 'assets/images/favicon.png']
+        const relativeFilePath = path.relative(directory, filePath);
+
+        // Convert destination file path to a POSIX path for uploading.
+        // Swift expects / as path separator and relativeFilePath will contain \\ on Windows.
+        // https://docs.openstack.org/python-openstackclient/pike/cli/man/openstack.html
+        const relativeFilePathPosix = relativeFilePath
+          .split(path.sep)
+          .join(path.posix.sep);
+
+        // The / delimiter is intentional since it represents the cloud storage and not the local file system.
+        const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+        const destination = `${entityRootDir}/${relativeFilePathPosix}`; // Swift container file relative path
+
+        const readStream = fs.createReadStream(filePath, 'utf8');
```

Comment:
please move the stream creation into the limiter below, so it doesn't create an unbounded number of streams at startup that aren't being read until later